File: packages/javascript/lib/node/file/getFileName.ts
import { posix, win32 } from 'path';

export function getFileName(filePath: string, withExtension = false): string {
	const removeExtension = !withExtension && !filePath.endsWith('/') && !filePath.endsWith('\\');

	// For Windows path
	if (filePath.indexOf('/') === -1 && filePath.indexOf('\\') !== -1) {
		return win32.basename(filePath, removeExtension ? win32.extname(filePath) : undefined) || '';
	}

	return posix.basename(filePath, removeExtension ? posix.extname(filePath) : undefined) || '';
}

File: packages/python/qsu/file/getFileName.py
import posixpath
import ntpath


def getFileName(filePath: str, withExtension: bool = False) -> str:
	removeExtension = (
		not withExtension
		and not filePath.endswith('/')
		and not filePath.endswith('\\')
	)

	# For Windows path
	if filePath.find('/') == -1 and filePath.find('\\') != -1:
		base = ntpath.basename(filePath.rstrip('\\'))
		if removeExtension:
			ext = ntpath.splitext(base)[1]
			if ext:
				base = base[: -len(ext)]
		return base or ''

	base = posixpath.basename(filePath.rstrip('/'))
	if removeExtension:
		ext = posixpath.splitext(base)[1]
		if ext:
			base = base[: -len(ext)]
	return base or ''
